realm-annotations-processor/src/main/java/io/realm/processor/RealmProcessor.java

```diff
@@ -30,15 +31,73 @@
 
 import io.realm.annotations.RealmClass;
 
+/**
+ * The RealmProcessor is responsible for creating the plumbing that connects the RealmObjects to a Realm. The process
+ * for doing so is summarized below and then described in more detail.
+ *
+ *
+ * DESIGN GOALS
+ *
+ * The processor should support the following design goals:
+ *
+ * - Minimize reflection.
+ * - As much as possible can be obfuscated.
+ * - Library projects must be able to use Realm without interfering with app code.
+ * - App code must be able to use model classes provided by library code.
+ * - It should work for app developers out of the box (ie. put the burden on the library developer)
+ *
+ *
+ * SUMMARY
+ *
+ * 1 ) Create proxy classes for all classes marked with @RealmClass. They are named <modelClass>RealmProxy.java
```

Comment:
Mention annotations in the documentation probably require a backslash to get JavaDoc to work.